### PR TITLE
Add machine_util.py for tuning optimal benchmark settings

### PR DESCRIPTION
**Pull request by wconstab**
Applies some configuration settings automatically, but verifies more.

Future work 
- utilize the query part from inside benchmark to embed queried values in json results
- additional optimizations like nohz, other nvidia settings, disabling IRQ
- add temp/clock query for gpu; try querying nvidia-smi performance mode

### CODE REVIEW

File: scripts/machine_util.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            for chunk in chunks:
+                if '-' in chunk:
+                    start, end = chunk.split('-')
+                    for cpu in range(start, end+1):
+                        isolcpus.add(cpu)
+                else:
+                    isolcpus.add(int(chunk))
+    return list(isolcpus)
+
+def get_nvidia_graphics_clock(device_id=0):
```
Comment (xuzhao9) [inline scripts/machine_util.py:88]: Does it make sense to get all available graphics clock by default?

**Review by xuzhao9 — APPROVED** [review]
LGTM with small comments.

File: scripts/machine_util.py (hunk 1 above)
Comment (wconstab) [reply scripts/machine_util.py:88]: yep, i have already made that change actually. I am adding more stuff still.  I'll update the PR in a little bit.

**Review by blaquezzz — APPROVED** [review]

File: torchbenchmark/util/machine_config.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    with open(sysfile, 'r') as f:
+        return f.read()
+
+def write_sys_file(sysfile: Path, content: str):
+    print(f"Write {content} to {sysfile}")
+    with open(sysfile, 'w') as f:
+        f.write(content)
+
+def check_intel_turbo_state(turbo_file='/sys/devices/system/cpu/intel_pstate/no_turbo'):
+    return int(read_sys_file(turbo_file))
```
[the remark this answers is not in the document]
Comment (wconstab) [reply torchbenchmark/util/machine_config.py:27]: oh yea good point.

**Pull request merged**